packages/audioplayers/lib/src/audioplayer.dart

```diff
@@ -253,10 +259,13 @@ class AudioPlayer {
     // First stop and release all native resources.
     await release();
 
-    final futures = <Future>[];
-    if (!_playerStateController.isClosed) {
-      futures.add(_playerStateController.close());
-    }
+    final futures = <Future>[
+      if (!_playerStateController.isClosed)
+        _playerStateController.close(),
+      if (_onPlayerCompleteStreamSubscription != null)
+        _onPlayerCompleteStreamSubscription!.cancel()
+    ];
+
```

Comment:
Also no need to check for null, if it's always initialized:

```suggestion
    final futures = <Future>[
      if (!_playerStateController.isClosed) _playerStateController.close(),
      _onPlayerCompleteStreamSubscription.cancel()
    ];

```